schema/verify/functions/date_or_null.sql

```diff
@@ -0,0 +1,47 @@
+-- Verify seattleflu/schema:functions/date_or_null on pg
+
+begin;
+
+create temporary table tests (
+    input text,
+    expected date
+);
+
+copy tests from stdin with (delimiter ';');
+2020-01-17;2020-01-17
+2020/01/17;2020-01-17
+2020.01.17;2020-01-17
+2020-01-17 00:00:00;2020-01-17
+2020/01/17 00:00:00;2020-01-17
+2020.01.17 00:00:00;2020-01-17
+Jan 17, 2020;2020-01-17
+January 17, 2020;2020-01-17
+Fri 17 Jan 12:24:00 2020 PDT;2020-01-17
+2020-01;\N
+2;\N
+na;\N
+\N;\N
+\.
+
+do $$
+    declare
+        test tests;
+    begin
+        for test in table tests loop
+            if test.expected is distinct from date_or_null(test.input) then
+                raise exception 'date_or_null(%) returned %, expected %',
+                    test.input, date_or_null(test.input), test.expected;
+            end if;
+        end loop;
+
+        begin
+            perform date_or_null('2020-01-50');
+            raise exception 'date_or_null() did not raise an error on out of range date';
+        exception
+            when datetime_field_overflow then
+                null; -- expected!
+        end;
+    end
+$$;
```

Comment:
Great to see these tests! I don't know about you, but I find it really satisfying to write tests of simple functions like this. :nerd_face:

Reply:
Yes. _Very_ satisfying 🤓